jib-core/src/main/java/com/google/cloud/tools/jib/tar/TarStreamBuilder.java

```diff
@@ -89,6 +91,7 @@ public void addByteEntry(byte[] contents, String name) {
   public void addBlobEntry(Blob blob, long size, String name) {
     TarArchiveEntry entry = new TarArchiveEntry(name);
     entry.setSize(size);
+    entry.setModTime(DEFAULT_MODIFICATION_TIME.getEpochSecond());
```

Comment:
I realized that this isn't enough. The `addByteEntry()` above also needs to set the timestamp. Both `addBlobEntry()` and `addByteEntry()` are used in `ImageTarball` to create a tar image stream.

However, we have layers of abstraction in our codebase with a hierarchy, so it doesn't feel right to make `TarStreamBuilder` to get into the business of deciding the timestamp of tar entries. This is a general class to build a tar stream. What I think we should do is to change the signature of the method to `addBlobEntry(Blob blob, long size, String name, Instant modTime)`, and have `ImageTarball` pass in the time.

Reply:
> I think this needs to be solved in ImageTarBall's calls to TarStreamBuilder (#addByteEntry and #addBlobEntry). Those methods should be allowing a timestamp parameter.

Oh, just realized @loosebazooka already suggested moving this to `ImageTarball` 😂